agent/consultemplate.go

```diff
@@ -0,0 +1,65 @@
+package agent
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/pkg/errors"
+
+	consultemplateconfig "github.com/hashicorp/consul-template/config"
+	"github.com/hashicorp/consul-template/manager"
+)
+
+type ConsulTemplateConfig struct {
+	Source      string
+	Destination string
+}
+
+func StartConsulTemplate(a *Agent, c *ConsulTemplateConfig) (*manager.Runner, error) {
```

Comment:
to be more consistent with the other functions in agent, it could be 

```
func (a *Agent) StartConsulTemplate(c * ConsulTemplateConfig) (*manager.Runner, error) {
```

instead.

Reply:
I'm not sure on this. This would state that this `StartConsulTemplate` method belongs to the `Agent` struct, and I created this code in a different go module to make it more explicit. It can be definitely done but to have all the pieces in order, it would require to move this methods to the `agent.go` module in my opinion (we don't want to have `agent` struct methods spread off in different files).
What do you think on this @stefanotorresi ?

Reply:
The package is still the same, so there is nothing really preventing you from adding more methods to the `Agent` struct. Putting methods in an a different file is not uncommon as a Go convention, so I wouldn't worry about that. In fact, I would argue that `agent/agent.go` is too crowded already.
One thing I would actually change is to make this method private, because it's not meant to be consumed outside the Agent itself.

This also means that this method would need to be first invoked by `agent.Start()` instead of being called directly by the command handler, as it is now.